File: devices/sources/teams.7.c
/*
* @@name:	teams.7
* @@type:	C
* @@operation:	run
* @@expect:	success
* @@version:	omp_6.0
*/
#include<stdio.h>
#include<omp.h>

int x;
#pragma omp declare target local(x)

int main() {
  x = 128;
  #pragma omp target
  x = 256;

  #pragma omp target
  #pragma omp teams num_teams(x) // Undefined behavior due to value of "x"
  if (omp_get_team_num() == 0){
    printf("%d\n", omp_get_num_teams()); 
  }

  return 0;
}

File: devices/sources/teams.7.f90
! @@name:	teams.7
! @@type:	F-free
! @@operation:	run
! @@expect:	success
! @@version:	omp_6.0
PROGRAM main
  USE omp_lib
  INTEGER :: x
  !$OMP DECLARE TARGET LOCAL(x)

   x = 128

  !$OMP TARGET
   x = 256
  !$OMP END TARGET 

  !$OMP TARGET
  !$OMP TEAMS NUM_TEAMS(x)        ! Undefined behavior due to value of 'x'
  IF (omp_get_team_num() == 0) THEN
    PRINT *, omp_get_num_teams()  
  END IF
  !$OMP END TEAMS
  !$OMP END TARGET

END PROGRAM main
